src/boards/nrf52840_m2/board.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Yihui Xiong for Makerdiary
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef _M2_H_
+#define _M2_H_
+
+#define _PINNUM(port, pin)    ((port)*32 + (pin))
+
+/*------------------------------------------------------------------*/
+/* LED
+ *------------------------------------------------------------------*/
+#define LEDS_NUMBER           1
+#define LED_PRIMARY_PIN       _PINNUM(0, 30)  // Red
+#define LED_DOCK_GREEN_PIN    _PINNUM(1, 7)   // Only available on the M.2 Dock
+#define LED_STATE_ON          0
+
+#define LED_RGB_RED_PIN       _PINNUM(0, 30)
+#define LED_RGB_GREEN_PIN     _PINNUM(0, 29)
+#define LED_RGB_BLUE_PIN      _PINNUM(0, 31)
+#define BOARD_RGB_BRIGHTNESS  0x404040
+/*------------------------------------------------------------------*/
+/* BUTTON
+ *------------------------------------------------------------------*/
+#define BUTTONS_NUMBER        2
+#define BUTTON_1              _PINNUM(0, 19)
+#define BUTTON_2              _PINNUM(1, 7)
+#define BUTTON_PULL           NRF_GPIO_PIN_PULLUP
+
+//--------------------------------------------------------------------+
+// BLE OTA
+//--------------------------------------------------------------------+
+#define BLEDIS_MANUFACTURER   "cmd.wtf"
+#define BLEDIS_MODEL          "PowerOverwhelmiing"
+
+//--------------------------------------------------------------------+
+// USB
+//--------------------------------------------------------------------+
+#define USB_DESC_VID            0x1915
```

Comment:
Do you have the permission from Nordic to use this VID/PID ?

Reply:
I'm not sure, but I'm guessing no. I just duplicated the data from Makerdiary's fork, in which they used this VID/PID. I see in some other boards it looks like you're sharing the VID/PID of the feather. Is that ideal for devices that don't have their own unique ones?

Reply:
I realize now I commited these changes with string I was using for my device I'm tinkering with and not the ones from MakerDiary. Somehow I managed to get my branches confused! Changed now, in 8ab7a41.

Still not sure what to do about the VID/PID though -- I've looked around and can't see any public listings Nordic keeps of PID's they've handed out for their VID.

Reply:
And as I continue to ramble on....

I've found these posts [here](https://devzone.nordicsemi.com/f/nordic-q-a/33851/nrf52840---usb-vendor-id-and-usb-product-id) and more specifically [this one](https://devzone.nordicsemi.com/f/nordic-q-a/30440/nrf24lu1-pid-vid/120663#120663), which makes it seem like Nordic allows open use of their VID, as long as you're shipping on a Nordic chip. They just don't track PIDs so they may collide, but that seems like a small issue for us!

Reply:
That will cause conflict in the future, it is still better to share a generic VID/PID bootloader than conflicting with other application mode. Since most bootloader behaves (and have the same usb interfaces). Therefore you have 2 options
1. Try to get circuitpython running on your board, and submit PR to its repo. You are eligible to get a free VID/PID from Adafruit. There is lots of PR/issues regarding this in circuipython. You can take a look there.
2. Wait for me to make some changes (with some internal discussion with Adafruit team) to decide which VID/PID can be used as generic one. 

Except for some early board sharing the VID/PID. Most of the boards that has unique here are donated one from Adafruit. It is relatively easy proess, you can try it out.

Reply:
So as I started looking around, it turns out it actually already is mainlined in circuitpython as of about a month ago it looks like: https://github.com/adafruit/circuitpython/blob/main/ports/nrf/boards/makerdiary_nrf52840_m2_devkit/mpconfigboard.mk

They're using a PID/VID that was granted to them by Seeed, it seems (0x2886/0xF001)!

So I looked at a couple of the adafruit boards between circuitpython and here. It seems they all use the same PID for the bootloader vs circuitpython's, except circuitpython's has the MSB 1 and the bootloader's has the MSB 0?

Should I see about asking adafruit for a bootloader-specific PID on the adafruit VID? Or should I instead try to get in touch with @makerdiary to see if perhaps Seeed had given them some other PIDs on the Seeed VID to use for bootloaders? I will tweet them with this comment just in case anyways!